### PR TITLE
feat: Add sync registration with storage

### PR DESCRIPTION
**Pull request by Elyniss**
Merge after #773

### CODE REVIEW

File: androidCore/sdk/src/main/kotlin/com/walletconnect/android/internal/common/di/DBUtils.kt [hunk 1]
```diff
@@ -9,6 +9,7 @@ object DBUtils {
     const val CHAT_SDK_DB_NAME = "WalletConnectV2_chat.db"
     const val PUSH_DAPP_SDK_DB_NAME = "WalletConnectV2_DappPush.db"
     const val PUSH_WALLET_SDK_DB_NAME = "WalletConnectV2_WalletPush.db"
+    const val SYNC_SDK_DB_NAME = "WalletConnectV2_sync.db"
```
Comment (Elyniss) [inline androidCore/sdk/src/main/kotlin/com/walletconnect/android/internal/common/di/DBUtils.kt:12]: What was necessary when adding a new db? I remember something about backup

File: sync/sdk/src/main/kotlin/com/walletconnect/sync/engine/use_case/SuspendUseCase.kt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package com.walletconnect.sync.engine.use_case
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+
+
+// todo: discuss if creating a separate for each scope use case is a good idea. If yes, then we might want to move this class to android core, for future refactor of other apis
```
Comment (Elyniss) [inline sync/sdk/src/main/kotlin/com/walletconnect/sync/engine/use_case/SuspendUseCase.kt:8]: Thoughts?

File: androidCore/sdk/src/main/kotlin/com/walletconnect/android/internal/common/di/DBUtils.kt (hunk 1 above)
Comment (TalhaAli00) [reply androidCore/sdk/src/main/kotlin/com/walletconnect/android/internal/common/di/DBUtils.kt:12]: You need to add these flags to the manifest of the module `android:fullBackupContent="@xml/backup_rules"` and `android:dataExtractionRules="@xml/backup_config"`. Examples in Push and Sign

File: sync/sdk/src/main/kotlin/com/walletconnect/sync/engine/use_case/SuspendUseCase.kt (hunk 2 above)
Comment (Elyniss) [reply sync/sdk/src/main/kotlin/com/walletconnect/sync/engine/use_case/SuspendUseCase.kt:8]: It is removed in #779

File: androidCore/sdk/src/main/kotlin/com/walletconnect/android/internal/common/di/DBUtils.kt (hunk 1 above)
Comment (Elyniss) [reply androidCore/sdk/src/main/kotlin/com/walletconnect/android/internal/common/di/DBUtils.kt:12]: Added in #779

**Review by jakubuid — APPROVED** [review]

**Review by TalhaAli00 — APPROVED** [review]

File: sync/sdk/src/main/kotlin/com/walletconnect/sync/storage/AccountsStorageRepository.kt [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import com.walletconnect.android.internal.common.model.AccountId
+import com.walletconnect.sync.common.model.Account
+import com.walletconnect.sync.common.model.Entropy
+import com.walletconnect.sync.data.dao.AccountsQueries
+
+internal class AccountsStorageRepository(private val accounts: AccountsQueries) {
+
+    suspend fun createAccount(account: Account) = with(account) {
+        // Only insert when account does not yet exists in db. Entropy will be the same so no need for multiple inserts/updates
+        if (doesAccountNotExists(accountId)) accounts.insertOrAbortAccount(accountId = accountId.value, entropy = entropy.value)
```
[the remark this answers is not in the document]
Comment (Elyniss) [reply sync/sdk/src/main/kotlin/com/walletconnect/sync/storage/AccountsStorageRepository.kt:12]: Yeah, you're right!

[the remark this answers is not in the document]
Comment (Elyniss) [reply sync/sdk/src/main/kotlin/com/walletconnect/sync/storage/AccountsStorageRepository.kt]: Removed: [`e52ee5c` (#775)](https://github.com/WalletConnect/WalletConnectKotlinV2/pull/775/commits/e52ee5c66f0faa446b3f6646895ab8b73f451329)

**Pull request merged**